framework/include/utils/RankFourTensor.h

```diff
@@ -66,6 +66,12 @@ template <typename T>
 class RankFourTensorTempl
 {
 public:
+  /// Dimensionality of rank-four tensor
+  static constexpr unsigned int N = LIBMESH_DIM;
```

Comment:
Just a comment, but we need to get away from this madness of tying our tensor sizes to LIBMESH_DIM. They just need to be hardcoded to 3 because they won't work any other way.